core/arch/interp.c

```diff
@@ -4196,6 +4223,12 @@ build_bb_ilist(dcontext_t *dcontext, build_bb_t *bb)
         }
 #endif
     }
+    else if (dcontext->forged_exception_addr == bb->start_pc) {
+        /* Might have been cleared by client_process_bb. */
```

Comment:
Sorry, my comment may have been not clear : bb->exit_type gets every time there is a client cleared by `client_process_bb`

```
    /* we set to NULL to have a default of fall-through */
    bb->exit_target = NULL;
    bb->exit_type = 0;
```

Another solution may be to change `client_process_bb`